### PR TITLE
[14.0][FIX] account_product_fiscal_classification: apply taxes if a default classification is specified using ir.default

### PR DESCRIPTION
**Pull request by kevinkhao**

### CODE REVIEW

File: account_product_fiscal_classification/models/product_template.py [hunk 1]
```diff
@@ -112,24 +112,31 @@ def fields_view_get(
                 result["arch"] = etree.tostring(doc)
         return result
 
+    def _apply_classification_taxes(self):
+        # update or replace 'taxes_id' and 'supplier_taxes_id'
+        classification = self.fiscal_classification_id
+        tax_vals = {
+            "supplier_taxes_id": [
+                (6, 0, [x.id for x in classification.sudo().purchase_tax_ids])
```
Comment (dreispt) [inline account_product_fiscal_classification/models/product_template.py]: Simpler like this?
```suggestion
                (6, 0, classification.sudo().purchase_tax_ids.ids)
```

File: account_product_fiscal_classification/models/product_template.py [hunk 2]
```diff
@@ -112,24 +112,31 @@ def fields_view_get(
...
+            ],
+            "taxes_id": [(6, 0, [x.id for x in classification.sudo().sale_tax_ids])],
```
Comment (dreispt) [inline account_product_fiscal_classification/models/product_template.py]:
```suggestion
            "taxes_id": [(6, 0, classification.sudo().sale_tax_ids.ids)],
```

File: account_product_fiscal_classification/models/product_template.py (hunk 1 above)
Comment (kevinkhao) [reply account_product_fiscal_classification/models/product_template.py]: indeed, thanks

**Comment by dreispt** [comment]
@legalsylvain You're the maintainer here. Can you review please?

**Comment by legalsylvain** [comment]
Hi @dreispt : thanks for pinging me. a pity that oca bot doesn't ping me. the feature was working before, AFAIK, right ?

@kevinkhao : I don't understand your patch. when ir.default is defined, the values should be in the vals, in the ``create(self, vals)`` part. so it should work without that patch ? 
or did I missed something ?

EDIT : 
how I see...
```
            # special case: using ir.default for fiscal_classification_id means
            #   value is not in vals, but is actually present on record
```
weird corner case but oK.


Just a question : I don't understand the name of ``create_mode_fiscal_sync``. what does ``sync`` means ?
something like ``create_mode_fiscal_classification`` seems better. (or did I missed something)...

**Pull request closed without merging**